clang/tools/driver/RemoteCache/CMakeLists.txt

```diff
@@ -1,17 +1,18 @@
 include_directories($<TARGET_PROPERTY:llvmRemoteCacheKVProto,BINARY_DIR>)
 include_directories($<TARGET_PROPERTY:llvmRemoteCacheCASProto,BINARY_DIR>)
 
-# Create an object library instead of a whole static archive.
-add_library(clangRemoteCache OBJECT EXCLUDE_FROM_ALL
-  Client.cpp
+set(LLVM_LINK_COMPONENTS
+  RemoteCacheCASProto
+  RemoteCacheKVProto
 )
-llvm_update_compile_flags(clangRemoteCache)
 
-target_link_libraries(clangRemoteCache
-  llvmRemoteCacheCASProto
-  llvmRemoteCacheKVProto
-)
-add_dependencies(clangRemoteCache
-  llvmRemoteCacheCASProto
-  llvmRemoteCacheKVProto
+add_clang_library(clangRemoteCache
+  Client.cpp
 )
+if (TARGET obj.clangRemoteCache)
```

Comment:
Do we always generate objlib? If so, this should work but what if the check returns false?

Reply:
I've seen other places where such a check exists; I'm not sure if that target is always there or there's some dependency on the CMake version or the generator you used, but I added the check to be conservative.

I expect without `obj.<target>` that the files will be generated ahead of compiling, since this is the case for the files of `llvm-remote-cache-test` and there's no `obj.` target for it.